Samples/2.2/GenericHost/src/SiloHost/Program.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using HelloWorld.Grains;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Hosting;
+using Orleans;
+using Orleans.Configuration;
+using Orleans.Hosting;
+
+namespace OrleansSiloHost
+{
+    public class Program
+    {
+        public static async Task<int> Main(string[] args)
+        {
+            try
+            {
+                var host = new HostBuilder()
+                    .AddOrleans(orleansBuilder =>
+                    {
+                        orleansBuilder
+                            .UseLocalhostClustering()
+                            .Configure<ClusterOptions>(options =>
+                            {
+                                options.ClusterId = "dev";
+                                options.ServiceId = "HelloWorldApp";
+                            })
+                            .Configure<EndpointOptions>(options => options.AdvertisedIPAddress = IPAddress.Loopback)
+                            .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(HelloGrain).Assembly).WithReferences())
+                            .ConfigureLogging(logging => logging.AddConsole());
+                    })
+                    .Build();
+                await host.RunAsync();
+
+                await host.StopAsync();
```

Comment:
Perhaps this should wait for the user to hit Ctrl+C before terminating?

Reply:
Just remove the `StopAsync()`. The host handles it for you.

Reply:
Fixed